inc/packages/namespace.php

```diff
@@ -0,0 +1,582 @@
+			$$manual_file_error = get_site_option( "altis_search_package_error_manual-{$file_type_string}", null );
+		} else {
+			$$uploaded_file_status = get_option( "altis_search_package_status_uploaded-{$file_type_string}" );
+			$$manual_file_status = get_option( "altis_search_package_status_manual-{$file_type_string}" );
+			$$uploaded_file_error = get_option( "altis_search_package_error_uploaded-{$file_type_string}", null );
+			$$manual_file_error = get_option( "altis_search_package_error_manual-{$file_type_string}", null );
+		}
+
+		$$uploaded_file_var = get_package_path( "uploaded-{$file_type_string}", $for_network );
+		$$manual_file_var = get_package_path( "manual-{$file_type_string}", $for_network );
+
+		$$text_var = '';
+		if ( file_exists( $$manual_file_var ) ) {
+			// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+			$$text_var = file_get_contents( $$manual_file_var );
+		}
+
+		$$file_date_var = false;
+		if ( file_exists( $$uploaded_file_var ) ) {
+			$$file_date_var = filemtime( $$uploaded_file_var );
+		}
+	}
+
+	$did_update = isset( $_GET['did_update'] );
+
+	if ( $for_network ) {
+		$errors = get_site_option( 'altis_search_config_error', false );
+	} else {
+		$errors = get_option( 'altis_search_config_error', false );
+	}
+
+	if ( ! empty( $errors ) && is_array( $errors ) ) {
+		$errors = array_map( function ( $error ) {
+			return new WP_Error( $error['code'], $error['message'] );
+		}, $errors );
+	}
+
+	include __DIR__ . '/templates/config.php';
+
+	if ( ! empty( $errors ) ) {
+		if ( $for_network ) {
+			delete_site_option( 'altis_search_config_error' );
+		} else {
+			delete_option( 'altis_search_config_error' );
+		}
+	}
+}
+
+/**
+ * Sets the search config error message.
+ *
+ * @param WP_Error $error The error object to add.
+ * @param bool $for_network Whether this should be added as a network level error.
+ */
+function add_error_message( WP_Error $error, bool $for_network = false ) {
+	static $errors = [];
+	$errors[] = $error;
+	$errors = array_map( function ( $error ) {
+		if ( ! is_wp_error( $error ) ) {
+			return $error;
+		}
+		return [
+			'code' => $error->get_error_code(),
+			'message' => $error->get_error_message(),
+		];
+	}, $errors );
+
+	if ( is_network_admin() || $for_network ) {
+		update_site_option( 'altis_search_config_error', $errors );
+	} else {
+		update_option( 'altis_search_config_error', $errors );
+	}
+}
+
+/**
+ * Returns the base directory for storing Elasticsearch packages.
+ *
+ * @return string
+ */
+function get_packages_dir() : string {
+	$path = WP_CONTENT_DIR . '/uploads/es-packages';
+
+	/**
+	 * Filter the path at which Elasticsearch package files are stored.
+	 *
+	 * @param string $path Absolute directory path, defaults to wp-content/es-packages.
+	 */
+	$path = apply_filters( 'altis.search.packages_dir', $path );
+
+	return $path;
+}
+
+/**
+ * Get a package file path. Returns the path if the file exists
+ * or null on failure.
```

Comment:
Short and long descriptions should be split. Also, the indicated return value is wrong anyway. Suggest removing second sentence entirely.

```suggestion
 * Get a package file path.
```